tango_simlib/tango_sim_generator.py

```diff
@@ -282,63 +297,82 @@ def initialize_dynamic_attributes(self):
+                )
+
+            return attribute
+
+        def _configure_attribute_default_properties(self, attribute, quantity_meta_data):
+            attribute_properties = UserDefaultAttrProp()
+            for prop, prop_value in quantity_meta_data.items():
+                # NB: Calling 'set_enum_labels' or setting the 'enum_labels' results
+                # in a error, and we do not need to do anyway as DevEnum attributes are
+                # handled by the `add_static_attribute` method.
+                if prop == "enum_labels":
                     continue
+                if hasattr(attribute_properties, prop):
+                    setattr(attribute_properties, prop, prop_value)
                 else:
-                    # The return value of rwType is a string and it is required as a
-                    # PyTango data type when passed to the Attr function.
-                    # e.g. 'READ' -> tango._tango.AttrWriteType.READ
-                    rw_type = meta_data["writable"]
-                    rw_type = getattr(AttrWriteType, rw_type)
-                    # Add a try/except clause when creating an instance of Attr class
-                    # as PyTango may raise an error when things go wrong.
-                    try:
-                        attr = Attr(attribute_name, attr_dtype, rw_type)
-                    except Exception as e:
-                        self._not_added_attributes.append(attribute_name)
-                        MODULE_LOGGER.info(
-                            "Attribute %s could not be added dynamically"
-                            " due to an error raised %s.",
-                            attribute_name,
-                            str(e),
-                        )
-                        continue
-                    attr_props = UserDefaultAttrProp()
-                    for prop in meta_data.keys():
-                        attr_prop_setter = getattr(attr_props, "set_" + prop, None)
-                        # CAVEAT (MTO): breaks silently; take note when debugging
-                        if attr_prop_setter:
-                            attr_prop_setter(meta_data[prop])
-                        else:
-                            MODULE_LOGGER.info(
-                                "No setter function for " + prop + " property"
-                            )
-                    attr.set_default_properties(attr_props)
-
-                    if rw_type in (AttrWriteType.READ, AttrWriteType.READ_WITH_WRITE):
-                        self.add_attribute(attr, self.read_attributes)
-                    elif rw_type == AttrWriteType.WRITE:
-                        self.add_attribute(attr, w_meth=self.write_attributes)
-                    elif rw_type == AttrWriteType.READ_WRITE:
-                        self.add_attribute(
-                            attr, self.read_attributes, self.write_attributes
-                        )
-
                     MODULE_LOGGER.info(
```

Comment:
Should this be changed to debug?  The dish simulator startup has previously been very noisy.